python/tempo/io.py

```diff
@@ -31,11 +30,15 @@ def write(
     df = tsdf.df
     ts_col = tsdf.ts_col
     partitionCols = tsdf.partitionCols
+
+    # TODO: this assumption of "event_time" column name is not appropriate
     if optimizationCols:
         optimizationCols = optimizationCols + ["event_time"]
     else:
         optimizationCols = ["event_time"]
 
+    # TODO: improve this logic. We should be checking for optimizationCols, not
+    # DATABRICKS_RUNTIME_VERSION
     useDeltaOpt = os.getenv("DATABRICKS_RUNTIME_VERSION") is not None
 
     view_df = df.withColumn("event_dt", sfn.to_date(sfn.col(ts_col))).withColumn(
```

Comment:
I think this was intended to check for a Databricks runtime before open-sourcing of OPTIMIZE and Z-ORDER. Do you think a better intent would be to verify a Delta version that supports these commands is installed?

For ref:
https://github.com/delta-io/delta/issues/927
https://github.com/delta-io/delta/issues/1134

Reply:
I think the attempt to run OPTIMIZE should maybe be dependent on whether or not `optimizationCols` is passed in? And if an improper Delta version is installed then a warning should be thrown.

Reply:
I think a near-term goal should be a re-think of all of this code, and designing a more generalizable solution for reading & writing TSDFs from/to various sources. 

In the meantime, I think we can trust the user to know if their platform supports this optimization or not. `optimizationCols` is None by default. If they specify some set of columns, we can go ahead and attempt to run the z-order operation. Worst case they get an exception the first time they try this, and (hopefully) they either stop using that arg, or they upgrade their environment to one that supports z-ordering.